Modules/Forum/classes/class.ilForumNotification.php

```diff
@@ -300,7 +300,9 @@ public static function getCachedNodeData($ref_id) : array
             $node_data = array_filter($node_data, static function (array $forum_node) use ($DIC, $ref_id) : bool {
                 // filter out forum if a grp lies in the path (#0027702)
                 foreach ($DIC->repositoryTree()->getNodePath($forum_node['child'], $ref_id) as $path_node) {
-                    if ((int) $path_node['child'] !== (int) $ref_id && $path_node['type'] === 'grp') {
+                    $notRootNode = (int) $path_node['child'] !== (int) $ref_id;
```

Comment:
Thanks for introducing the variables here, which make the condition more understandable.